### PR TITLE
Vue docs

### PR DESCRIPTION
**Pull request by pax-k**
- implemented a VueJS Todo demo app `examples/todo-vue`
- `useCoState()` id argument now accepts either a plain value or a computed value
- added docs to `/docs/project-setup/vue`

### CODE REVIEW

File: examples/todo-vue/src/views/HomeView.vue [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+watch(folders, (loadedFolders) => {
+  if (selectedFolder.value) return;
+  selectedFolder.value = loadedFolders?.[0] || undefined;
+});
+
+const selectFolder = (folder: Folder) => {
+  selectedFolder.value = folder;
+};
+
+const createFolder = async () => {
```
Comment (aeplay) [inline examples/todo-vue/src/views/HomeView.vue]: Does this need to be async?

File: examples/todo-vue/src/views/HomeView.vue [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    },
+    { owner: group },
+  );
+
+  // Add the folder to the list of folders. This change will be synced to all connected clients.
+  folders.value?.push(newFolder);
+  newFolderName.value = "";
+};
+
+const deleteFolder = async (folderId: ID<Folder> | undefined) => {
```
Comment (aeplay) [inline examples/todo-vue/src/views/HomeView.vue]: async?

File: examples/todo-vue/src/views/HomeView.vue [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    folders.value.splice(index, 1);
+  }
+
+  if (selectedFolder.value?.id === folderId) {
+    selectedFolder.value = folders.value[0] || null;
+  }
+};
+
+// Todo handlers
+const createTodo = async () => {
```
Comment (aeplay) [inline examples/todo-vue/src/views/HomeView.vue]: async?

File: examples/todo-vue/src/views/HomeView.vue [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+  );
+
+  // Add the todo to the list of todos. This change will be synced to all connected clients.
+  // toRaw is used to get the plain object from the reactive object, because the plain object is already proxied by Jazz.
+  // otherwise it will throw an error.
+  toRaw(selectedFolder.value)?.items?.push(newTodo);
+  newTodoTitle.value = "";
+};
+
+const deleteTodo = async (todoId: ID<ToDoItem> | undefined) => {
```
Comment (aeplay) [inline examples/todo-vue/src/views/HomeView.vue]: async?

File: packages/jazz-vue/src/createJazzVueApp.ts [hunk 5]
```diff
@@ -244,7 +251,7 @@ export function createJazzVueApp<Acc extends Account>({
 
   function useCoState<V extends CoValue, D>(
     Schema: CoValueClass<V>,
-    id: ID<V> | undefined,
+    id: ComputedRef<ID<V> | undefined>,
```
Comment (aeplay) [inline packages/jazz-vue/src/createJazzVueApp.ts]: Is this the only way to make it reactive to `id`? I don't know enough about Vue but the call site looks very elaborate...

Comment (pax-k) [reply packages/jazz-vue/src/createJazzVueApp.ts]: A plain `id` reference is not tracked and will not reactively update useCoState(). 

In this example folders will be undefined:
```
const computedFoldersId = me.value?.root?.folders?.id
const folders = useCoState(FolderList, computedFoldersId, [{ items: [{}] }]);
```

In this example we track changes to `me.value?.root?.folders?.id` and `useCoState()` tracks changes to computedFoldersId: 
```
const computedFoldersId = computed(() => me.value?.root?.folders?.id);
const folders = useCoState(FolderList, computedFoldersId, [{ items: [{}] }]);
```

Comment (pax-k) [reply packages/jazz-vue/src/createJazzVueApp.ts]: Now `id` is `MaybeRef<ID<V> | undefined>`

It accepts either a plain value, or a ref.

**Pull request merged**